src/msg/async/rdma/Infiniband.cc

```diff
@@ -267,6 +267,98 @@ int Infiniband::QueuePair::init()
   return 0;
 }
 
+void Infiniband::QueuePair::wire_gid_to_gid(const char *wgid, ib_cm_meta* cm_meta_data)
+{
+  char tmp[9];
+  uint32_t v32;
+  int i;
+
+  for (tmp[8] = 0, i = 0; i < 4; ++i) {
```

Comment:
Ha! GID is a 128-bit value, which in its turn valid 128-bit IPv6 address (RFC 2373).  Instead of doing such hairy conversion init_ntop() can be easily invoked, e.g.:

	inet_ntop(AF_INET6, &my_dest[i].gid, gid, sizeof(gid));

Reply:
inet_ntop API will not keep 0 in gid.
fe80::268a:7ff:fe60:90e1 should be fe80:0000:0000:0000:268a:07ff:fe60:90e1

Reply:
@rouming I'll simplify the implement to send/recv connection management metadata.
The main changes are:
1) transfer cm meta data obj plain object directly instead of transferring "char msg[TCP_MSG_LEN]"
2) remove wire_gid_to_gid/gid_to_wire_gid since we'll analysing the recv/send data in cm meta data structure directly.

Reply:
Done. Wait for updating PR

Reply:
> inet_ntop API will not keep 0 in gid.

that's fine, if you use the opposite conversion routine inet_pton() then it is fine. 

!!! BUT !!!! that will break protocol compatibility, also what you've suggested (transferring object instead of a string, what obviously should be done from the very beginning).  So, seems better to keep it as is. Now nothing can be done if you do not want to break protocol compatibility.

Reply:
inet_pton should work fine (decode 0) if we use inet_ntop.
I want to abandon the original method to transfer GID in string format.
Currently, I've implemented the change to transfer the plain object instead of string at the very beginning. I think it's easy to be maintained and read.

Reply:
As I said you break the protocol compatibility. So you won't be able to connect clients to OSDs with different versions.

Reply:
@rouming 
Got your point. I'll keep compatibility in this PR, i.e. keep "TCP_MSG_LEN" format here.